src/main/java/io/pinecone/clients/AsyncIndex.java

```diff
@@ -846,6 +848,178 @@ public ListenableFuture<DescribeIndexStatsResponse> describeIndexStats(Struct fi
+    @Override
+    public ListenableFuture<ListResponse> list(String namespace) {
+        validateListEndpointParameters(namespace, null, null, null, true, false, false, false);
+        ListRequest listRequest = ListRequest.newBuilder().setNamespace(namespace).build();
+        return asyncStub.list(listRequest);
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * <p>Example:
+     *  <pre>{@code
+     *     import io.pinecone.proto.ListResponse;
+     *     import com.google.common.util.concurrent.Futures;
+     *     import com.google.common.util.concurrent.ListenableFuture;
+     *
+     *     ...
+     *
+     *      ListenableFuture<ListResponse> futureResponse = asyncIndex.list("example-namespace", 10);
+     *      ListResponse asyncListResponse = Futures.getUnchecked(futureResponse);
+     *  }</pre>
+     */
+    @Override
+    public ListenableFuture<ListResponse> list(String namespace, Integer limit) {
+        validateListEndpointParameters(namespace, null, null, limit, true, false, false, true);
+        ListRequest listRequest = ListRequest.newBuilder().setNamespace(namespace).setLimit(limit).build();
+        return asyncStub.list(listRequest);
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * <p>Example:
+     *  <pre>{@code
+     *     import io.pinecone.proto.ListResponse;
+     *     import com.google.common.util.concurrent.Futures;
+     *     import com.google.common.util.concurrent.ListenableFuture;
+     *
+     *     ...
+     *
+     *     ListenableFuture<ListResponse> futureResponse = asyncIndex.list("example-namespace", "prefix-");
+     *     ListResponse asyncListResponse = Futures.getUnchecked(futureResponse);
+     *  }</pre>
+     */
+    @Override
+    public ListenableFuture<ListResponse> list(String namespace, String prefix) {
+        validateListEndpointParameters(namespace, prefix, null, null, true, true, false, false);
+        ListRequest listRequest = ListRequest.newBuilder().setNamespace(namespace).setPrefix(prefix).build();
+        return asyncStub.list(listRequest);
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * <p>Example:
+     *  <pre>{@code
+     *     import io.pinecone.proto.ListResponse;
+     *     import com.google.common.util.concurrent.Futures;
+     *     import com.google.common.util.concurrent.ListenableFuture;
+     *
+     *     ...
+     *
+     *     ListenableFuture<ListResponse> futureResponse = asyncIndex.list("example-namespace", "prefix-", 10);
+     *     ListResponse asyncListResponse = Futures.getUnchecked(futureResponse);
+     *  }</pre>
+     */
+    @Override
+    public ListenableFuture<ListResponse> list(String namespace, String prefix, Integer limit) {
+        validateListEndpointParameters(namespace, prefix, null, limit, true, true, false, true);
+        ListRequest listRequest = ListRequest.newBuilder().setNamespace(namespace).setPrefix(prefix).
+                setLimit(limit).build();
+        return asyncStub.list(listRequest);
+    }
```

Comment:
I would consider changing the type of `limit` to be an `int`. We can keep it as an `Integer` in the `validateListEndpointParameters` method to allow method overloads to set it to null when it's not relevant.

Reply:
Ah ty! I still am not used to `int` vs `Integer`, thanks!